crates/db_schema/src/impls/post.rs

```diff
@@ -372,6 +372,35 @@ impl PostRead {
     .await
     .with_lemmy_type(LemmyErrorType::CouldntMarkPostAsRead)
   }
+
+  pub async fn mark_many_as_read(
+    pool: &mut DbPool<'_>,
+    post_ids: &[PostId],
+    person_id: PersonId,
+  ) -> LemmyResult<usize> {
```

Comment:
There is no need to have a separate method `mark_as_read()`. Or at least you can change that one to call mark_many_as_read(), instead of duplicating the whole logic.